gbpservice/nfp/lib/nfp_log_helper.py

```diff
@@ -0,0 +1,144 @@
+                    else ''),
+                '04_service_chain_id': (
+                    ('ServiceChainID:' +
+                     nf['service_chain_id'])
+                    if nf.get('service_chain_id')
+                    else ''),
+                '05_service_chain_instance_id': (
+                    ('ServiceInstanceID:' +
+                     nf['id'])
+                    if nf.get('id')
+                    else ''),
+                '06_service_type': (
+                    ('ServiceType:' + service_type)
+                    if service_type
+                    else ''),
+                '07_service_provider': (
+                    ('ServiceProvider:' + service_provider)
+                    if service_provider
+                    else ''),
+            }
+        elif log_meta_data and log_meta_data != '':
+            """Reorganize the log_meta_data string along with new fields
+               like EventCategory, Events, Level etc
+               e.g log_meta_dat = '[TenantID:tid, ServiceChainID:scid,
+                                ServiceInstanceID:siid, ServiceType:FIREWALL,
+                                ServiceProvider:vyos]'
+            """
+
+            string = log_meta_data[1:-1].replace(" ", "")
+            nf = dict(kv.split(":") for kv in string.split(","))
+
+            log_info_nf_content = {
+                '03_tenant_id': (
+                    ('TenantID:' + nf['TenantID'])
+                    if nf.get('TenantID')
+                    else ''),
+                '04_service_chain_id': (
+                    ('ServiceChainID:' +
+                     nf['ServiceChainID'])
+                    if nf.get('ServiceChainID')
+                    else ''),
+                '05_service_chain_instance_id': (
+                    ('ServiceInstanceID:' +
+                     nf['ServiceInstanceID'])
+                    if nf.get('ServiceInstanceID')
+                    else ''),
+                '06_service_type': (
+                    ('ServiceType:' + nf['ServiceType'])
+                    if nf.get('ServiceType')
+                    else ''),
+                '07_service_provider': (
+                    ('ServiceProvider:' + nf['ServiceProvider'])
+                    if nf.get('ServiceProvider')
+                    else ''),
+            }
+
+        log_info_content.update(log_info_nf_content)
+        log_info_content = OrderedDict(sorted(log_info_content.items(),
+                                              key=lambda t: t[0]))
+
+        log_msg = '['
+        for content in log_info_content:
+            if log_info_content.get(content):
+                log_msg += log_info_content.get(content) + ', '
+        log_msg = log_msg[:-2]
+        if log_msg:
+            log_msg += '] - '
+
+        return log_msg
+    except Exception as err:
+        LOG.error(_LE("Error while generating LOG. %s."
+                      % str(err).capitalize()))
+        raise err
```

Comment:
This will go as unhandled exception, instead of raising exception  we can return empty string.